scripts/repo-variable-writer.go

```diff
@@ -0,0 +1,284 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+type Project struct {
+	Name          string   `json:"name"`
+	Organization  string   `json:"organization"`
+	SubComponents []string `json:"sub_components"`
+}
+
+type Projects struct {
+	Projects []Project `json:"projects"`
+}
+
+const (
+	LocProjMeta             string = "../projects/project.json"
+	LocWriteUpdatedProjects string = "/tmp/updates.json"
+)
+
+func toPtr[T any](v T) *T {
+	return &v
+}
+
+func handleHTTPCall(method string, url string, timeOut time.Duration,
+	body io.Reader, headers map[string]string) (*http.Response, error) {
+
+	client := &http.Client{
+		Timeout: timeOut,
+	}
+
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+
+	for k, v := range headers {
+		req.Header.Add(k, v)
+	}
+
+	return client.Do(req)
+}
+
+func loadProjectMetadata(locProj string) (resp *Projects, err error) {
+	_b, err := os.ReadFile(locProj)
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(_b, &resp)
+	return
+}
+
+func fetchLatestRelease(org, proj string) (*string, error) {
+	url := fmt.Sprintf(
+		"https://api.github.com/repos/%s/%s/releases/latest", org, proj)
+
+	resp, err := handleHTTPCall(
+		http.MethodGet,
+		url,
+		time.Minute, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Status code was not 200, got: %v", resp.StatusCode)
+	}
+
+	var release struct {
+		TagName string `json:"tag_name"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&release); err != nil {
+		return nil, fmt.Errorf("failed deserialize response body: %v", err)
+	}
+
+	slog.Info("Latest Release", "Proj", proj, "Org", org, "Ver", release.TagName)
+
+	return toPtr[string](release.TagName), nil
+}
+
+type GithubVariable struct {
+	gh_token     string
```

Comment:
```suggestion
	githubToken     string
```

Nit but lets use camel case for all the vars